moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -649,39 +657,34 @@ std::optional<PoseCommand> Servo::toPlanningFrame(const PoseCommand& command, co
 KinematicState Servo::getCurrentRobotState() const
 {
   moveit::core::RobotStatePtr robot_state = planning_scene_monitor_->getStateMonitor()->getCurrentState();
-  const moveit::core::JointModelGroup* joint_model_group =
-      robot_state->getJointModelGroup(servo_params_.move_group_name);
-  const auto joint_names = joint_model_group->getActiveJointModelNames();
-
-  KinematicState current_state(joint_names.size());
-  current_state.joint_names = joint_names;
-  robot_state->copyJointGroupPositions(joint_model_group, current_state.positions);
-  robot_state->copyJointGroupVelocities(joint_model_group, current_state.velocities);
-  robot_state->copyJointGroupAccelerations(joint_model_group, current_state.accelerations);
-
-  return current_state;
+  return extractRobotState(robot_state, servo_params_.move_group_name);
 }
 
 std::pair<bool, KinematicState> Servo::smoothHalt(const KinematicState& halt_state)
 {
   bool stopped = false;
   auto target_state = halt_state;
-  const KinematicState current_state = getCurrentRobotState();
 
-  const size_t num_joints = current_state.joint_names.size();
-  for (size_t i = 0; i < num_joints; ++i)
+  // set target velocity
+  target_state.velocities *= 0.0;
+
+  // apply smoothing: this will change target position/velocity to make slow down gradual
+  doSmoothing(target_state);
+
+  // scale velocity in case of obstacle
+  target_state.velocities *= collision_velocity_scale_;
+
+  for (size_t i = 0; i < (size_t)halt_state.positions.size(); ++i)
```

Comment:
Do you need to cast the size here? If so, consider using `static_cast` instead.

Reply:
It’s only needed because of the warning from comparing signed with unsigned numbers. I think I’d prefer to make the indexing variable a long instead.